mleap-tensor/src/main/scala/ml/combust/mleap/tensor/Tensor.scala

```diff
@@ -66,30 +75,12 @@ case class DenseTensor[T](values: Array[T],
   }
 
   override def get(indices: Int *): Option[T] = {
-    var i = 0
-    var dimI = 1
-    var n = indices.head
-    var tail = indices.tail
-    while(i < tail.size) {
-      var ti = dimI
-      var tn = tail.head
-      tail = tail.tail
-      while(ti < dimensions.size) {
-        tn *= dimensions(ti)
-        ti += 1
-      }
-      dimI += 1
-      i += 1
-      n += tn
-    }
```

Comment:
After reading this [blog post](https://eli.thegreenplace.net/2015/memory-layout-of-multi-dimensional-arrays), your new code makes sense to me:

Each time you increment a particular array index, you must traverse the memory addresses for all the faster-changing dimensions that come in between. It makes sense to me that the length of this traversal is equal to the product of all the faster-changing dimension sizes. Since you initialize `n = index.last`, it seems that the final dimension must be the fastest changing one, making your code row-major.

The old code confuses me though... Since they initialize `n = indices.head`, it would seem that the first dimension is the fastest-changing one, making it column-major. But the loop that looks like this:
```
while(ti < dimensions.size) {
    tn *= dimensions(ti)
    ti += 1
}
```
Seems to be multiplying `tn` by all the slower-changing dimensions that come after it. Instead, I think it should be multiplying together all the faster-changing dimensions that come **before** it.

Is the legacy code incorrect?

Reply:
Yes. The legacy code has bug.
Here is the code the verify.
describe("Test Legacy DenseTensor Get") {
  def legacy_get(indices: Seq[Int], dimensions: Seq[Int]) = {
    //This is the legacy get method from DenseTensor
    var i = 0
    var dimI = 1
    var n = indices.head
    var tail = indices.tail
    while (i < tail.size) {
      var ti = dimI
      var tn = tail.head
      tail = tail.tail
      while (ti < dimensions.size) {
        tn *= dimensions(ti)
        ti += 1
      }
      println(tn)
      dimI += 1
      i += 1
      n += tn
      println(n)
    }
    n
  }
  it("Fortran layout") {
    assert(legacy_get(Seq(0, 0), Seq(3, 4)) == 0)
    assert(legacy_get(Seq(2 , 3), Seq(3, 4)) == 3*3+2)
    assert(legacy_get(Seq(1, 1), Seq(3, 4)) == 1*3+1)
    assert(legacy_get(Seq(1, 1, 1), Seq(3, 4, 5)) == 1*4*3 + 1*3+ 1)
    assert(legacy_get(Seq(2, 3, 4), Seq(3, 4, 5)) == 4*4*3 + 3*3+ 2)
  }
}

Reply:
@ancasarb This is the main part that we need you to verify
If we're correct, then everybody's `DenseTensor` indexing has been wrong this entire time

That would be very surprising

Reply:
thanks @voganrc, i am taking a look at this now.

Reply:
Hi @ancasarb I create a branch from master in my forked repo.
[test_nd_tensor](https://github.com/austinzh/mleap/tree/test_nd_tensor
)
Feel free to check it out and run `sbt mleap-tensor/test`. 
Here I attached a small piece of the logs.
[mleap-tensor-test.log](https://github.com/combust/mleap/files/6901973/mleap-tensor-test.log)